### PR TITLE
docs: how it works

### PR DESCRIPTION
**Pull request by bwanglzu**
Added how it works section and the documentation structure:

see it lively here: https://ft-docs-how-it-works--jina-docs.netlify.app/docs-how-it-works/

please review:
1. index page
2. how it works section
3. documentation structure

C254CD90-AE41-433C-B77C-C765C47E1438

### CODE REVIEW

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# How Finetuner Works
+
+## Contrastive Metric Learning 👓
+
+From the algorithmic perspective,
+**Finetuner** leverage contrastive metric learning approach to improve your model.
+How it works?
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:7]: I think "How it works?" is somehow grammatically incorrect, I would phrase it "How does it work?"

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+### Step 1: Convert your model into an Embedding Model
+
+Finetuner interpret your model architecture,
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:11]: "interprets"

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+remove the default *head*, apply *pooling* and freeze layers not necessary to train.
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:12]: "removes ... applies ... freezes layers that do not need to be trained."

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+For example, if your pre-trained model is trained for cat-dog classification.
+Finetuner is going to remove the cat-dog classifier and turn your model into an *embedding model*.
+
+This *embedding model* do not make predictions and output a probability,
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:16]: "This *embedding model* does not make predictions or outputs a probability, but instead outputs a *feature vector* to represent your data.

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+but output a *feature vector* to represent your data.
+
+### Step 2: Triplets Construction and Training on-the-fly
+
+Finetuner "looks" into the label of your training data.
+Each `Document` and `Document`s share the same `finetuner_label` is considered as a *Positive*.
+In the meanwhile, each `Document` and `Documents` share a different `finetuner_label` is considered as a *Negative*.
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:23]: "which share"

Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:23]: also "which share"

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# How Finetuner Works
+
+## Contrastive Metric Learning 👓
+
+From the algorithmic perspective,
+**Finetuner** leverage contrastive metric learning approach to improve your model.
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:6]: levarages a

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+while push the `Document`s belongs to the different class away from each other.
+
+
+
+## Cloud-based Fine-tuning ⛅
+
+From engineering perspective,
+we have hide all the complexity of machine learning algorithms,
+resources setup (such as GPU),
+all you need to do is to decide your backbone model and prepare your training data.
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:37]: "From an engineering perspective, we have hidden all the complexity of machine learning algorithms and resources configuration (such as GPU). All you need ... "

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+Once you logged into Jina Ecosystem with `finetuner.login()`,
+Finetuner will push your training data into our *Cloud Artifact Storage* (only visible to you).
+At the same time, we will spin-up an isolated computational resource
+with proper memory, CPU, GPU dedicated for your fine-tune job.
+
+Once fine-tuning is done, Finetuner will again push your `tuned_model` to the *Cloud Artifact Storage*
+and available for you to pull it back to your machine,
```
Comment (violenil) [inline docs/1_how_it_works/1_1_how_it_works.md:45]: "and make it available for you ... "

File: docs/1_how_it_works/1_2_difference.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# Difference between Finetuner and Others
+
+There are several fancy machine learning libraries available,
+what makes Finetuner unique?
+
+## Finetuner care about the quality of Embeddings 🧬
+
+Finetuner is not designed to make better classification,
```
Comment (violenil) [inline docs/1_how_it_works/1_2_difference.md:8]: "Finetuner is not designed to improve classification, ..."

File: docs/1_how_it_works/1_2_difference.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+sentiment analysis or object detection.
+Finetuner only care about the quality of embeddings.
+The fine-tuned model only produces embeddings.
```
Comment (violenil) [inline docs/1_how_it_works/1_2_difference.md:11]: Finetuner only cares about the quality of embeddings, and this is what the fine-tuned model will produce.

File: docs/1_how_it_works/1_2_difference.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+Given a query `Document` embeded into `embeddings`,
```
Comment (violenil) [inline docs/1_how_it_works/1_2_difference.md:13]: "embedded"

File: docs/1_how_it_works/1_2_difference.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+you can compare the similarity/distance of the query `Document`s against all indexed (embeded) `Document`s in your storage backend.
+
+
+## Finetuner is dedicated to optimize search task 🎯
+
+Finetuner helps you boost your search system performance on different modalities of data.
+Text-to-text, image-to-image, text-to-image and more is coming!
+
+Search performance depends on a lot of factors.
+Internally we have conducted a lot of experiments on varies tasks,
```
Comment (violenil) [inline docs/1_how_it_works/1_2_difference.md:23]: " ... various tasks"

File: docs/1_how_it_works/1_2_difference.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+**Finetuner is able to boost 20%-45% of precision@k and recall@k**.
+You can also observe significant performance improvement on other search metrics,
+such as mean recipal rank (mRR) or normalized discounted cumulative gain (nDCG).
+
+## Finetuner is extremely easy to use 🚀
+
+Finetuner give user the flexibility to choose machine learning hyper-parameters,
+while all these parameters are `Optional`.
+
+If you do not have machine learning background,
```
Comment (violenil) [inline docs/1_how_it_works/1_2_difference.md:37]: "If you do not have a machine learning background ..."

**Comment by JohannesMessner** [comment]
I have some high-level feedback:

- I really like the structure, the user can immediately see what actions they will have to take to successfully fine tune their model
- Headings should follow sentence case, according to our guidelines, so e.g. "Choose backbone model" instead of "Choose Backbone Model"
- Using `backticks` in heading loogs a bit weird in my opinion
- The "how finetuner works" page makes sense to me. We should just make sure that we explain things at the adequate technical level for the target user. I am not 100% sure what that level is, just something that we should always think about :)

Really good backbone, this will become a nice piece of documentation!

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+but instead outputs a feature vector to represent your data
+
+### Step 2: Triplets Construction and Training on-the-fly
+
+Finetuner "looks" into the label of your training data.
+Each `Document` and `Document`s share the same ****`finetuner_label` is considered as a *Positive*.
+In the meanwhile, each `Document` and `Documents` which share a different `finetuner_label` is considered as a *Negative*.
+
+During model fine-tuning, Finetuner is creating *Triplets*  ``(anchor, positive, negative)`` on-the-fly.
+Finetuner then uses the triplets to perform training,
```
Comment (jupyterjazz) [inline docs/1_how_it_works/1_1_how_it_works.md:26]: I think this part needs a bit more clarity.

Some suggestions:

Finetuner works on labeled data. It expects a `DocumentArray` consisting of `Documents` where each one contains `finetuner_label` corresponding to a class of a specific training example.

During the fine-tuning, Finetuner creates *Triplets*  ``(anchor, positive, negative)`` on-the-fly.
For each *anchor*, which can be any training example, Finetuner looks for a `Document` with the same `finetuner_label` (*positive*), and a `Document` with a different `finetuner_label` (*negative*).

File: docs/1_how_it_works/1_2_difference.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+## Easy to use 🚀
+
+Finetuner gives user the flexibility to choose machine learning hyper-parameters,
+while all these parameters are `Optional`.
+
+If you do not have a machine learning background,
+not a problem.
+As was stated before, you only need to send us the training data organized as a [DocumentArray](https://docarray.jina.ai/).
+In case you do not know which backbone to choose,
+use `finetuner.list_models()` to let Finetuner suggest a backbone model for you.
```
Comment (jupyterjazz) [inline docs/1_how_it_works/1_2_difference.md:45]: I think it's better to describe the models in the docs.
What about listing only model names with `list_models()` and having guides/suggestions in the docs?

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
+# How Finetuner Works
+
+## Contrastive Metric Learning 👓
+
+From an algorithmic perspective,
+**Finetuner** leverages a contrastive metric learning approach to improve your model.
+How does it work?
+
+### Step 1: Convert your model into an Embedding Model
```
[the remark this answers is not in the document]
Comment (bwanglzu) [reply docs/1_how_it_works/1_1_how_it_works.md:9]: we support this for most of the CNN models, in the Runner

[the remark this answers is not in the document]
Comment (bwanglzu) [reply docs/1_how_it_works/1_1_how_it_works.md:9]: ah I got you, good point, let me rephrase it a bit

**Comment by samsja** [comment]
I really like the docs good structure and content:

One comment : 
the `Cloud-based fine-tuning` part could used a schema. They are a lot of entities.

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+## Contrastive metric learning
+
+From an algorithmic perspective,
+**Finetuner** leverages a contrastive metric learning approach to improve your model.
+How does it work?
+
+### Step 1: Convert a model into an Embedding Model
+
+Finetuner interprets the backbone model architecture,
+removes the default *head*, applies *pooling* and freezes layers that do not need to be trained..
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:12]: one dot

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+For example, if your pre-trained model is trained for cat-dog classification,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:13]: for an image classification task (e.g. cats and dogs), Finetuner is going to remove the classification head (cat-dog classifier) and turn ...

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+Finetuner is going to remove the cat-dog classifier and turn your model into an *embedding model*.
+
+This embedding model does not make predictions or outputs a probability,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:16]: does not make predictions neither it outputs probabilities

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+but instead outputs a feature vector to represent your data.
+
+### Step 2: Triplet construction and training on-the-fly
+
+Finetuner works on labeled data.
+It expects a `DocumentArray` consisting of `Document`s where each one contains `finetuner_label` corresponding to a class of a specific training example.
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:22]: each one contains a `finetuner_label` tag corresponding to the class of a specific ...

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+Finetuner looks for a `Document` with the same `finetuner_label` (positive),
+and a `Document` with a different `finetuner_label` (negative).
+The objective is to pull `Document`s which belong to the same class together,
+while pushing the `Document`s which belong to a different class away from each other.
+
+
+## Cloud-based fine-tuning
+
+From an engineering perspective,
+we have hidden all the complexity of machine learning algorithms and resources configuration (such as GPU).
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:36]: resource configuration (such as GPUs)

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+all you need to do is to decide your backbone model and prepare your training data.
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:37]: **All you need** to do is decide **on**

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+
+Once you logged into Jina Ecosystem with `finetuner.login()`,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:39]: Once you are logged into the

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+Finetuner will push your training data into our *Cloud Artifact Storage* (only visible to you).
+At the same time, we will spin-up an isolated computational resource
+with proper memory, CPU, GPU dedicated for your fine-tune job.
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:42]: fine-tuning job

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+
+Once fine-tuning is done, Finetuner will again push your `tuned_model` to the *Cloud Artifact Storage*
+and make it available for you to pull it back to your machine,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_1_how_it_works.md:45]: machine. That's it!

File: docs/1_how_it_works/1_2_difference.md [hunk 26]
```diff
@@ -0,0 +1,45 @@
+# Difference between Finetuner and other frameworks
+
+There are several fancy machine learning libraries available,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_2_difference.md:3]: libraries out there, so what makes ...

File: docs/1_how_it_works/1_2_difference.md [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+what makes Finetuner unique?
+
+## Care about the quality of embeddings
+
+Finetuner is not designed to improve classification,
+sentiment analysis or object detection.
+Finetuner cares the quality of embeddings,
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_2_difference.md:10]: cares about the quality of the

File: docs/1_how_it_works/1_2_difference.md [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+and this is what the fine-tuned model will produce.
+
+Given a query `Document` embedded into `embeddings`,
+you can compare the similarity/distance of the query `Document`s against all indexed (embedded) `Document`s in your storage backend.
+
+
+## Dedicated to optimizing your search task
+
+Finetuner helps you boost your search system performance on different modalities of data:
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_2_difference.md:19]: on different use cases

File: docs/1_how_it_works/1_2_difference.md [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+You can also observe significant performance improvement on other search metrics,
+such as mean recipal rank (mRR) or normalized discounted cumulative gain (nDCG).
+
+## Easy to use
+
+Finetuner gives user the flexibility to choose machine learning hyper-parameters,
+while all these parameters are `Optional`.
+
+If you do not have a machine learning background,
+not a problem.
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_2_difference.md:42]: don't worry about it.

File: docs/1_how_it_works/1_2_difference.md [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+As was stated before, you only need to send us the training data organized as a [DocumentArray](https://docarray.jina.ai/).
```
Comment (gmastrapas) [inline docs/1_how_it_works/1_2_difference.md:43]: you only need to provide

**Review by gmastrapas — CHANGES_REQUESTED** [review]

**Comment by gmastrapas** [comment]
Also add a line in CHANGELOG, have you merged the changes from main? I think this is why tests are failing

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 31]
```diff
@@ -0,0 +1,51 @@
+# How Finetuner works
+
+## Contrastive metric learning
```
Comment (jemmyshin) [inline docs/1_how_it_works/1_1_how_it_works.md:3]: What if users have their own labeled data? Will we still use contrastive metrics learning instead of supervised learning?

File: docs/1_how_it_works/1_1_how_it_works.md [hunk 32]
```diff
@@ -0,0 +1,51 @@
...
+
+This embedding model does not make predictions or outputs a probability,
+but instead outputs a feature vector to represent your data.
+
+### Step 2: Triplet construction and training on-the-fly
+
+Finetuner works on labeled data.
+It expects a `DocumentArray` consisting of `Document`s where each one contains `finetuner_label` corresponding to the class of a specific training example.
+
+During the fine-tuning, Finetuner creates Triplets `(anchor, positive, negative)` on-the-fly.
```
Comment (jemmyshin) [inline docs/1_how_it_works/1_1_how_it_works.md:24]: I think we also support `CLIPLoss` right?

**Review by gmastrapas — APPROVED** [review]

File: docs/1_how_it_works/1_2_difference.md [hunk 33]
```diff
@@ -0,0 +1,45 @@
+# Difference between Finetuner and other frameworks
+
+There are several fancy machine learning libraries out there,
+so what makes Finetuner unique?
+
+## Care about the quality of embeddings
+
+Finetuner is not designed to improve classification,
```
Comment (jemmyshin) [inline docs/1_how_it_works/1_2_difference.md:8]: I personally like this way: "Finetuner is not aiming at boosting the performance of classification ... **directly**"

Comment (jemmyshin) [reply docs/1_how_it_works/1_2_difference.md:8]: Since embeddings can be used in any downstream tasks

**Comment by jemmyshin** [comment]
The structure of the documentation looks good to me.

File: docs/1_how_it_works/1_1_how_it_works.md (hunk 31 above)
Comment (bwanglzu) [reply docs/1_how_it_works/1_1_how_it_works.md:3]: by default it's supervised

File: docs/1_how_it_works/1_1_how_it_works.md (hunk 32 above)
Comment (bwanglzu) [reply docs/1_how_it_works/1_1_how_it_works.md:24]: yes, and it is still leveraging the triplets
